analyzers/src/SonarAnalyzer.Common/Rules/ParametersCorrectOrderBase.cs

```diff
@@ -20,139 +20,57 @@
-
-            if (nameColonIdentifier == null)
-            {
-                return new PositionalArgumentIdentifier(identifierName, argument);
-            }
-
-            return new NamedArgumentIdentifier(identifierName, argument, nameColonIdentifier.Value.Text);
-        }
-
-        private class ArgumentIdentifier
-        {
-            protected ArgumentIdentifier(string identifierName, TArgumentSyntax argumentSyntax)
-            {
-                IdentifierName = identifierName;
-                ArgumentSyntax = argumentSyntax;
-            }
-
-            public string IdentifierName { get; }
-            public TArgumentSyntax ArgumentSyntax { get; }
-        }
-
-        private class PositionalArgumentIdentifier : ArgumentIdentifier
-        {
-            public PositionalArgumentIdentifier(string identifierName, TArgumentSyntax argumentSyntax)
-                : base(identifierName, argumentSyntax)
-            {
-            }
-        }
-
-        private class NamedArgumentIdentifier : ArgumentIdentifier
-        {
-            public NamedArgumentIdentifier(string identifierName, TArgumentSyntax argumentSyntax, string declaredName)
-                : base(identifierName, argumentSyntax)
-            {
-                DeclaredName = declaredName;
-            }
-
-            public string DeclaredName { get; }
-        }
+                    if (!c.IsRedundantPrimaryConstructorBaseTypeContext()
+                        && Language.Syntax.ArgumentList(c.Node) is { Count: >= 2 } argumentList // there must be at least two arguments to be able to swap
+                        && Language.MethodParameterLookup(c.Node, c.SemanticModel) is var methodParameterLookup)
+                    {
+                        foreach (var argument in argumentList)
+                        {
+                            // Example void M(int x, int y) <- p_x and p_y are the parameter
+                            // M(y, x); <- a_y and a_x are the arguments
+                            if (methodParameterLookup.TryGetSymbol(argument, out var parameterSymbol) // argument = a_x and parameterSymbol = p_y
+                                && parameterSymbol is { IsParams: false }
+                                && ArgumentName(argument) is { } argumentName // "x"
+                                && !MatchingNames(parameterSymbol, argumentName)  // "x" != "y"
+                                && Language.Syntax.NodeExpression(argument) is { } argumentExpression
+                                && c.Context.SemanticModel.GetTypeInfo(argumentExpression).ConvertedType is { } argumentType
+                                // is there another parameter that seems to be a better fit (name and type match): p_x
+                                && methodParameterLookup.MethodSymbol.Parameters.FirstOrDefault(p => MatchingNames(p, argumentName) && argumentType.DerivesOrImplements(p.Type)) is { IsParams: false }
```

Comment:
```suggestion
                                && methodParameterLookup.MethodSymbol.Parameters.FirstOrDefault(p => MatchingNames(p, argumentName) 
                                && argumentType.DerivesOrImplements(p.Type)) is { IsParams: false }
```

Reply:
Why?

Reply:
you inlined that AND condition
, all the others are listed one after the other